Lib/xml/dom/xmlbuilder.py

```diff
@@ -247,10 +247,11 @@ def _create_opener(self):
 
     def _guess_media_encoding(self, source):
         info = source.byteStream.info()
-        if "Content-Type" in info:
-            for param in info.getplist():
-                if param.startswith("charset="):
-                    return param.split("=", 1)[1].lower()
+        # import email.message
+        # assert isinstance(info, email.message.Message)
+        for ctp_name, ctp_value in info.get_params(()):
+            if ctp_name == 'charset':
+                return ctp_value.lower()
```

Comment:
Why not use `get_param()`?
```suggestion
        charset = info.get_param('charset')
        if charset is not None:
            return charset.lower()
        return None
```

Reply:
I think the answer is that I stopped looking at methods of `Message` after I found the one that looked like a direct replacement of the old code. This makes sense to me.